src/mlpack/core/data/split_data.hpp

```diff
@@ -179,13 +180,15 @@ void StratifiedSplit(const arma::Mat<T>& input,
  * @param shuffleData If true, the sample order is shuffled; otherwise, each
  *       sample is visited in linear order. (Default true.)
  */
-template<typename T, typename U>
+template<typename T, typename LabelsType,
+         typename = std::enable_if_t<arma::is_Row<LabelsType>::value ||
```

Comment:
Hmm, `is_arma_type` may be sufficient here (since then this could probably work for `Cube`s and `SpMat`s too).  Let me know what you think of that idea.

Reply:
Yeah you are correct. I will update this in the next commit :+1:

Reply:
I apologize, I overlooked here. The restrictor is on `LabelsType`. And I have never came across 3D labels. They are always either 1D or 2D, but I have never seen 3D labels.
So, I don't think we should allow for `Cube`s. Although they would work well with this implementation, but in my opinion they are not required.
`SpMat`s can be used in the case when the labels are one hot encoded.
So, I would say we should only add `SpMat` here. Also, I wanted to know if there is any restrictor for `SpMat` ?
Correct me if I still missed something :-)

Reply:
I think it would be fine to use `is_arma_type`.  No need to restrict so that users can't use `Cube`s.  (Why not, after all?  Just because we haven't seen it doesn't mean someone wouldn't need to use it like that.)

Reply:
Hey @rcurtin , I will change it to `is_arma_type`. Currently I am burdened by my college exams, so I am unable to find some time to do this. I will try to implement all the changes that you suggested within a week or two and also try to finish my other WIP PR also :)

Reply:
No worries!  Good luck on the exams. :+1: I'll keep an eye out for updates to this PR so that whenever you do have the time I can hopefully provide a quick review.